lib/rubocop/cop/lint/literal_in_interpolation.rb

```diff
@@ -29,7 +28,9 @@ def on_dstr(node)
         private
 
         def special_string?(node)
-          node.type == :str && !node.loc.respond_to?(:begin)
+          # handle strings like __FILE__
+          (node.type == :str && !node.loc.respond_to?(:begin)) ||
+            node.loc.expression.is?('__LINE__')
```

Comment:
__LINE__ is not a string, so handling it in a method called `special_string?` is kinda confusing.